src/core/src/engine.rs

```diff
@@ -504,6 +508,23 @@ fn validator_is_claiming_too_early( service: &mut Poet2Service )->bool
 
 }
 
+fn create_context() -> Result<lmdb::LmdbContext, CliError> {
+    let path_config = config::get_path_config();
+    let statestore_path = &path_config.data_dir.join(config::get_filename());
+
+    lmdb::LmdbContext::new(statestore_path, 1, None)
+        .map_err(|err| CliError::EnvironmentError(format!("{}", err)))
+}
+
+fn open_statestore(ctx: &lmdb::LmdbContext) -> Result<ConsensusStateStore, CliError> {
```

Comment:
How do we close state store and release the context created.

Reply:
Will scope in context cleanup if need in later iteration.